lib/sync/sync.go

```diff
@@ -213,6 +224,33 @@ func (wg *loggedWaitGroup) Wait() {
 	}
 }
 
+type closeWaitGroup struct {
+	sync.WaitGroup
+	closed   bool
+	closeMut sync.RWMutex
+}
+
+func NewCloseWaitGroup() CloseWaitGroup {
```

Comment:
This doesn't seem generic enough to warrant inclusion and exporting in lib/sync, could be just a private type in package backend like the releaser thing?